### ISSUE TITLE
Moving a list only works once

### ISSUE THREAD

**Issue by birefringence** [stefan-niedermann/nextcloud-deck#889]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

Moving a list left or right only works once in the app, then no further moves are possible.

1. Select a board with more than 2 lists already present.
2. Create a new list.
3. Click on 'Move list to the left'.
4. Click on 'Move list to the left' again.

I would expect the new list to be in the third position from the right, but only the first move works. The list remains stuck in the second position from the right.

**Versions**
 - Nextcloud: 21.0.0
 - Nextcloud Deck: 1.3.1
 - Nextcloud Android: 3.15.1
 - Nextcloud Android Deck: 1.15.6

**Smartphone (please complete the following information):**
- Device: Sony Xperia XA2 (H3113)
- Android-Version: 10 (Lineage OS 17.1)
- App-Store:
  - [x] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [ ] F-Droid
  - [ ] Huawei AppGallery

**Comment by stefan-niedermann** [comment]
Hi @birefringence 
i can not reproduce the described behavior, the lists can be moved properly to the left multiple times... Are you able to reproduce this constantly?

**Comment by birefringence** [comment]
I tried again. This time, I was able to move the list twice until it was stuck. Restarting the App does not help.
```
2021-04-19 14:11:59 INFO (MainActivity.java:458) onCreateStack() → Create Stack in account user@domain.de/nextcloud on board 3
(MainActivity.java:640) lambda$null$17$MainActivity() → java.lang.IllegalStateException: Could not generate tab title for position 3 because list size is only 4
	at it.niedermann.nextcloud.deck.ui.MainActivity.lambda$null$17$MainActivity(MainActivity.java:640)
	at it.niedermann.nextcloud.deck.ui.-$$Lambda$MainActivity$AjqKQNgxFe_o_le2HBd0hmAxGRI.getTitle(Unknown Source:4)
	at it.niedermann.nextcloud.deck.ui.MainActivity.lambda$null$18(MainActivity.java:644)
	at it.niedermann.nextcloud.deck.ui.-$$Lambda$MainActivity$eAwH7m6-geGqOPlpOrzE7P5vbec.onConfigureTab(Unknown Source:2)
	at com.google.android.material.tabs.TabLayoutMediator.populateTabsFromPagerAdapter(TabLayoutMediator.java:177)
	at com.google.android.material.tabs.TabLayoutMediator$PagerAdapterObserver.onChanged(TabLayoutMediator.java:287)
	at androidx.recyclerview.widget.RecyclerView$AdapterDataObservable.notifyChanged(RecyclerView.java:12253)
	at androidx.recyclerview.widget.RecyclerView$Adapter.notifyDataSetChanged(RecyclerView.java:7354)
	at it.niedermann.nextcloud.deck.ui.stack.StackAdapter.setStacks(StackAdapter.java:54)
	at it.niedermann.nextcloud.deck.ui.MainActivity.lambda$setCurrentBoard$21$MainActivity(MainActivity.java:626)
	at it.niedermann.nextcloud.deck.ui.-$$Lambda$MainActivity$WMXVvIH_d1iPefPxvngGLCVcRgk.onChanged(Unknown Source:4)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
	at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
	at androidx.lifecycle.Transformations$3.onChanged(Transformations.java:188)
	at androidx.lifecycle.MediatorLiveData$Source.onChanged(MediatorLiveData.java:152)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

(MainActivity.java:640) lambda$null$17$MainActivity() → java.lang.IllegalStateException: Could not generate tab title for position 3 because list size is only 4
	at it.niedermann.nextcloud.deck.ui.MainActivity.lambda$null$17$MainActivity(MainActivity.java:640)
	at it.niedermann.nextcloud.deck.ui.-$$Lambda$MainActivity$AjqKQNgxFe_o_le2HBd0hmAxGRI.getTitle(Unknown Source:4)
	at it.niedermann.android.tablayouthelper.TabLayoutHelper.onCreateTab(TabLayoutHelper.java:160)
	at it.niedermann.android.tablayouthelper.TabLayoutHelper.createNewTab(TabLayoutHelper.java:241)
	at it.niedermann.android.tablayouthelper.TabLayoutHelper.setTabsFromPagerAdapter(TabLayoutHelper.java:269)
	at it.niedermann.android.tablayouthelper.TabLayoutHelper.access$300(TabLayoutHelper.java:37)
	at it.niedermann.android.tablayouthelper.TabLayoutHelper$3.onChanged(TabLayoutHelper.java:100)
	at androidx.recyclerview.widget.RecyclerView$AdapterDataObservable.notifyChanged(RecyclerView.java:12253)
	at androidx.recyclerview.widget.RecyclerView$Adapter.notifyDataSetChanged(RecyclerView.java:7354)
	at it.niedermann.nextcloud.deck.ui.stack.StackAdapter.setStacks(StackAdapter.java:54)
	at it.niedermann.nextcloud.deck.ui.MainActivity.lambda$setCurrentBoard$21$MainActivity(MainActivity.java:626)
	at it.niedermann.nextcloud.deck.ui.-$$Lambda$MainActivity$WMXVvIH_d1iPefPxvngGLCVcRgk.onChanged(Unknown Source:4)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
	at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
	at androidx.lifecycle.Transformations$3.onChanged(Transformations.java:188)
	at androidx.lifecycle.MediatorLiveData$Source.onChanged(MediatorLiveData.java:152)
	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)

2021-04-19 14:11:59 DEBUG (DeckApplication.java:159) readCurrentStackId() → --- Read: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 6
2021-04-19 14:12:00 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":1001,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:00 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":1001,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:00 DEBUG (DeckApplication.java:159) readCurrentStackId() → --- Read: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 6
2021-04-19 14:12:00 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:00 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 6
2021-04-19 14:12:01 VERBOSE (BrandingUtil.java:64) getSecondaryForegroundColorDependingOnTheme() → Contrast ratio between brand color #F0BF46 and primary theme background is too low. Falling back to WHITE/BLACK as brand color.
2021-04-19 14:12:01 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:04 DEBUG (DeckApplication.java:159) readCurrentStackId() → --- Read: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:04 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:04 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Gekauft","boardId":3,"deletedAt":0,"lastModified":1618420309,"order":1001,"id":9,"ETag":"931666c5dc9ba4d9391e8a8957c91cc4"}
2021-04-19 14:12:04 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Gekauft","boardId":3,"deletedAt":0,"lastModified":1618420309,"order":1001,"id":9,"ETag":"931666c5dc9ba4d9391e8a8957c91cc4"}
2021-04-19 14:12:04 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":1000,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:04 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":1000,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:06 DEBUG (DeckApplication.java:159) readCurrentStackId() → --- Read: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:06 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:07 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Konkret","boardId":3,"deletedAt":0,"lastModified":1618412317,"order":1000,"id":8,"ETag":"17545450aa3122d43151678ce8e64c59"}
2021-04-19 14:12:07 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Konkret","boardId":3,"deletedAt":0,"lastModified":1618412317,"order":1000,"id":8,"ETag":"17545450aa3122d43151678ce8e64c59"}
2021-04-19 14:12:07 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:07 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:09 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:09 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:09 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:09 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:11 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:11 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:11 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:11 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:12 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 6
2021-04-19 14:12:12 DEBUG (DeckApplication.java:151) saveCurrentStackId() → --- Write: it.niedermann.nextcloud.deck.last_stack_for_board_1_3 → 22
2021-04-19 14:12:15 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:15 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:15 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:15 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:18 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:18 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Ideen","boardId":3,"deletedAt":0,"lastModified":1618262143,"order":999,"id":7,"ETag":"d5b868957605a3a48317683a32a9fb27"}
2021-04-19 14:12:18 VERBOSE (NextcloudDeserializer.java:29) deserialize() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}
2021-04-19 14:12:18 VERBOSE (JsonToEntityParser.java:582) parseStack() → {"title":"Entwurf","boardId":3,"deletedAt":0,"lastModified":0,"order":999,"id":24,"ETag":"cfcd208495d565ef66e7dff9f98764da"}

**Comment by stefan-niedermann** [comment]
Fix will be in version `1.18.3`, thank you for the report.

**Comment by desperateCoder** [comment]
And for responding!
